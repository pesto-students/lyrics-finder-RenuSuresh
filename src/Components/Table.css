.table__background {
  height: fit-content;
  background-color: #360000;
  background-image: linear-gradient(to right, #d88b17, #6e6726);
}
.table__div {
  display: flex;
  justify-content: center;
}
p {
  display: flex;
  justify-content: center;
  font-size: 40px;
  font-weight: 600;
  font-family: cursive;
  text-shadow: 0 0 14px #d88b17;
}
.table__table {
  display: flex;
  justify-content: center;
}
table {
  border: 1px solid #ccc;
  border-collapse: collapse;
  padding: 0;
  /* width: 60%; */
  table-layout: fixed;
  margin: 0 auto;
  min-width: fit-content;
  margin: 25px;
  /* box-shadow: 0 4px 8px 0 red, 0 4px 8px 0 red; */
  box-shadow: 4px 8px 0 rgba(0, 0, 0, 0), 0 6px 20px 0 rgba(237, 172, 141, 0.8);
}

table caption {
  font-size: 1.5em;
  margin: 0.5em 0 0.75em;
}

table tr {
  background-color: #f8f8f8;
  border: 1px solid #ddd;
  padding: 0.35em;
}

table th,
table td {
  padding: 0.625em;
}

table th {
  font-size: 0.85em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
}

@media screen and (max-width: 600px) {
  table {
    border: 0;
  }

  table caption {
    font-size: 1.3em;
  }

  table thead {
    border: none;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }

  table tr {
    border: 3px solid #ddd;
    display: block;
    margin-bottom: 0.625em;
  }

  table td {
    border: 1px solid #ddd;
    display: block;
    font-size: 0.8em;
    text-align: left;
  }

  table td::before {
    content: attr(data-label);
    float: left;
    font-weight: bold;
    text-transform: uppercase;
  }

  table td:last-child {
    border-bottom: 0;
  }
}

tbody tr:hover {
  background-color: #d6d6d6;
}
.table__info {
  display: flex;
  flex-direction: column;
}
.table__title {
  font-size: small;
}
.table__album {
  font-family: "Courier New", Courier, monospace;
  font-size: 17px;
  font-weight: 600;
}
.table__img__col {
  width: 15%;
}
.table__caption {
  width: auto;
  display: flex;
  justify-content: center;
  margin-top: 64px;
}
.table__caption > p {
  color: #eeeaf0;
  border-bottom: 3px double #ffffff;
}
.table__div__border {
  border: 5px double white;
  width: 60%;
}
.close {
  color: white;
  font-size: xxx-large;
  margin-right: 15px;
  opacity: 1;
}
.close:hover {
  color: #ffc766;
  outline: none;
}
