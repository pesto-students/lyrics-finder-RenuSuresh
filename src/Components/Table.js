import React, { useEffect, useState } from "react";
import "./Table.css";
import { getSuggest } from "../axios";
import LyricsModal from "./LyricsModal";
import Pagination from "./Pagination";
import BeatLoader from "react-spinners/BeatLoader";

function TableData({ currentGenre, getLyrics }) {
  return (
    <table>
      <tbody>
        {currentGenre.map((Element) => (
          <tr
            key={Element.id}
            onClick={() =>
              getLyrics(
                Element.artist.name,
                Element.title,
                Element.album.title,
                Element.artist.picture_medium
              )
            }
          >
            <td className="table__img__col">
              <img src={Element.album.cover_small} alt="album" />
            </td>
            <td className="table__info">
              <span className="table__album">{Element.album.title}</span>
              <span className="table__title">{Element.title}</span>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
}

function Table({ musicType, hideTable }) {
  const [genre, setGenre] = useState([]);
  const [artist, setArtist] = useState("");
  const [title, setTitle] = useState("");
  const [currentPage, setCurrentPage] = useState(1);
  const [genrePerPage] = useState(5);
  const [loading, setLoading] = useState(true);
  const [image, setImage] = useState("");
  const [albumTitle, setAlbumTitle] = useState("");

  // get current data
  const indexOfLastGenre = currentPage * genrePerPage;
  const indexOfFirstGenre = indexOfLastGenre - genrePerPage;
  const currentGenre = genre.slice(indexOfFirstGenre, indexOfLastGenre);

  // change page
  const paginate = (pageNumber) => setCurrentPage(pageNumber);

  useEffect(() => {
    async function fetchData() {
      const request = await getSuggest.get(musicType);

      setGenre(request.data.data);
      setLoading(false);
      return request;
    }
    fetchData();
  }, [musicType]);
  console.log("genre>>>", genre);
  const getLyrics = (artist, title, album_title, image) => {
    setTitle(title);
    setArtist(artist);
    setImage(image);
    setAlbumTitle(album_title);
    console.log("albumtitle>>>>", album_title);
  };

  return (
    <div className="table__div">
      <div className="table__div__border">
        <button type="button" className="close" onClick={hideTable}>
          &times;
        </button>
        <div className="table__caption">
          <p>{musicType}</p>
        </div>

        <div className="table__table">
          {loading ? (
            <BeatLoader loading={loading} color={"#ffffff"} />
          ) : (
            <TableData currentGenre={currentGenre} getLyrics={getLyrics} />
          )}
        </div>
        <Pagination
          genrePerPage={genrePerPage}
          totalGenre={genre.length}
          paginate={paginate}
        />
        {artist && (
          <LyricsModal
            artist={artist}
            title={title}
            image={image}
            album_title={albumTitle}
          />
        )}
      </div>
    </div>
  );
}

export default Table;
