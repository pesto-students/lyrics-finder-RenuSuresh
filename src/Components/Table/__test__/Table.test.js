import { configure, EnzymeAdapter } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { shallow } from "enzyme";
import Table from "../Table";
import Enzyme from "enzyme";
import {
  render,
  fireEvent,
  getByTestId,
  getByText,
} from "@testing-library/react";
import toJson from "enzyme-to-json";
configure({ adapter: new Adapter() });
const { mount } = Enzyme;

describe("Table", () => {
  test("snapshot <Table/>", () => {
    const TableTree = mount(<Table />);
    expect(toJson(TableTree)).toMatchSnapshot();
  });
  test("behaviour for the caption of the table", () => {
    const mock = jest.fn();
    const { container } = render(
      <Table musicType="popular" hideTable={mock} />
    );
    const tableMusicType = getByTestId(container, "table-musictype");
    expect(tableMusicType.textContent).toBe("popular");
  });
  test("close button", () => {
    const mock = jest.fn();
    const { container } = render(
      <Table musicType="popular" hideTable={mock} />
    );
    const close = getByTestId(container, "close-btn");
    console.log(close.textContent);
    expect(close.textContent).toBe("X");
  });
});
