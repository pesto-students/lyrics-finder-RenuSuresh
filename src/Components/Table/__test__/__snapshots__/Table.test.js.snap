// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table snapshot <Table/> 1`] = `
<Table>
  <div
    className="table__div"
  >
    <div
      className="table__div__border"
    >
      <TableData
        loading={true}
      >
        <button
          className="close"
          data-testid="close-btn"
          type="button"
        >
          Ã—
        </button>
        <div
          className="table__caption"
        >
          <p
            data-testid="table-musictype"
          />
        </div>
        <div
          className="table__table"
        >
          <Loader
            color="#ffffff"
            css=""
            loading={true}
            margin={2}
            size={15}
          >
            <EmotionCssPropInternal
              __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
              css={
                Array [
                  "",
                ]
              }
            >
              <span
                className="css-0"
              >
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1njukjo",
                      "next": Object {
                        "name": "animation-1bzpvkt",
                        "next": undefined,
                        "styles": "@keyframes animation-1bzpvkt{
  50% {transform: scale(0.75);opacity: 0.2}
  100% {transform: scale(1);opacity: 1}
}",
                      },
                      "styles": "
      display: inline-block;
      background-color: #ffffff;
      width: 15px;
      height: 15px;
      margin: 2px;
      border-radius: 100%;
      animation: animation-1bzpvkt 0.7s 0s infinite linear;
      animation-fill-mode: both;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <span
                    className="css-1njukjo"
                  />
                </EmotionCssPropInternal>
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
                  css={
                    Object {
                      "map": undefined,
                      "name": "mhyaug",
                      "next": Object {
                        "name": "animation-1bzpvkt",
                        "next": undefined,
                        "styles": "@keyframes animation-1bzpvkt{
  50% {transform: scale(0.75);opacity: 0.2}
  100% {transform: scale(1);opacity: 1}
}",
                      },
                      "styles": "
      display: inline-block;
      background-color: #ffffff;
      width: 15px;
      height: 15px;
      margin: 2px;
      border-radius: 100%;
      animation: animation-1bzpvkt 0.7s 0.35s infinite linear;
      animation-fill-mode: both;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <span
                    className="css-mhyaug"
                  />
                </EmotionCssPropInternal>
                <EmotionCssPropInternal
                  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
                  css={
                    Object {
                      "map": undefined,
                      "name": "1njukjo",
                      "next": Object {
                        "name": "animation-1bzpvkt",
                        "next": undefined,
                        "styles": "@keyframes animation-1bzpvkt{
  50% {transform: scale(0.75);opacity: 0.2}
  100% {transform: scale(1);opacity: 1}
}",
                      },
                      "styles": "
      display: inline-block;
      background-color: #ffffff;
      width: 15px;
      height: 15px;
      margin: 2px;
      border-radius: 100%;
      animation: animation-1bzpvkt 0.7s 0s infinite linear;
      animation-fill-mode: both;
    ",
                      "toString": [Function],
                    }
                  }
                >
                  <span
                    className="css-1njukjo"
                  />
                </EmotionCssPropInternal>
              </span>
            </EmotionCssPropInternal>
          </Loader>
        </div>
      </TableData>
      <Pagination
        genrePerPage={5}
        paginate={[Function]}
        totalGenre={0}
      >
        <nav
          className="pagination__nav"
        >
          <ul
            className="pagination"
          />
        </nav>
      </Pagination>
    </div>
  </div>
</Table>
`;
